src/checks/series/step_check.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{DataCache, Error, Flag};
+
+/// Number of leading values a [`DataCache`] must contain to QC all its
+/// intended values with step check
+pub const STEP_LEADING_PER_RUN: u8 = 1;
+
+/// Timeseries QC test that compares each observation against its immediate predecessor.
+///
+/// If the absolute value of the difference between the observed value and it's predecessor is
+/// greater than max, Flag::Fail will be returned for that observation, if greater than high,
+/// Flag::Warn, if either value if missing, Flag::DataMissing, else Flag::Pass.
+///
+/// Takes 2 datapoints, the second is the observation to be QCed, the first is needed to QC it.
+pub fn step_check(data: &[Option<f32>; 2], max: f32) -> Flag {
+    if data.contains(&None) {
+        return Flag::DataMissing;
+    }
+    let data: Vec<f32> = data.iter().map(|opt| opt.unwrap()).collect();
+
+    if (data[0] - data[1]).abs() > max {
+        return Flag::Fail;
+    }
```

Comment:
Probably it doesn't make any difference?
```suggestion
    if (data[0].unwrap() - data[1].unwrap()).abs() > max {
        return Flag::Fail;
    }
```

Reply:
done